### ISSUE TITLE
Quality of life improvements for uploads

### ISSUE THREAD

**Issue by edoyango** [WEHI-ResearchComputing/FileSenderCli#17]
Tried uploading 100x10GB files in a single command i.e.

```bash
filesender --base-url https://filesender.aarnet.edu.au/rest.php upload --username yang.e@wehi.edu.au --apikey $my_api_key --recipients $an_email testfiles/testfile-{1..100}
```

* subcommand help cannot be accessed without supplying `--base-url`. Expected behaviour would be that I can use `--help` for a subcommand without the `--base-url`.
* ctrl+C has to be spammed to stop an upload or doesn't cancel upload
* upload is very resource greedy - it tries to use up as much ram+CPU as it can. It looks like it's buffering as much as it can in mem?
    * ideally, defaults would be sensible e.g., 1 CPU 1GB RAM
    * User can then request more if they want to e.g. through flags. I guess that's what `--chunk-size` is supposed to do?
* `upload --chunk-size` has behaves unexpectedly
    * causes the upload to take awhile to start (slowly fills up RAM, whereas without chunksize, upload appears to start straightaway)
    * Not clear at all what the chunk-size units are and how it relates to resource usage
        * The docs say that I can use `--chunk-size` to reduce memory, but it's not clear how to choose the right chunk size.
* a progress bar of some kind would be nice.

**Comment by multimeric** [comment]
Thanks, this is good feedback. Regarding the resource utilisation, I think that #16 would solve the high memory and/CPU usage. You would for example add `--concurrent 4` which would stop the app from holding more than 4 chunks in memory at a time, and also using more than 4 threads. Although the thread using with async is a bit hard to pin down.

**Comment by multimeric** [comment]
The resource issues should have been resolved by #16. The other issues are still outstanding.
